pkg/kv/store_message.go

```diff
@@ -0,0 +1,62 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const PathDelimiter = "/"
+
+// StoreMessage protobuf generic implementation for kv.Store interface applicable for all data models
+type StoreMessage struct {
+	Store Store
+}
+
+func (s *StoreMessage) GetMsg(ctx context.Context, path string, msg protoreflect.ProtoMessage) error {
+	val, err := s.Store.Get(ctx, []byte(path))
+	if err != nil {
+		return fmt.Errorf("failed on Get. Path (%s): %w", path, err)
```

Comment:
Here and elsewhere

I think that the way errors are wrapped here may lead to long error lines with repetitive information, imagine the following possible error:
```
failed on Get. Path (this/is/some/very/long/repetitive/path): failed on Get. Path (this/is/some/very/long/repetitive/path): failed on Get. Path (this/is/some/very/long/repetitive/path): postgres failed to connect. Path (this/is/some/very/long/repetitive/path): tcp: no such host
```
Now it's repetitive because there are several calls to Get with path (gateway, storeMessage, KVStore and probably others I missed.)

I suggest the following (I think it's the golang approach)
```suggestion
		return fmt.Errorf("getting from store: %w", err)
```

If the path is relevant for the error, it should be added at the very start of the chain where the path has the most meaning.